src/core/qgsmaplayer.cpp

```diff
@@ -1301,6 +1301,28 @@ QString QgsMapLayer::loadDefaultMetadata( bool &resultFlag )
 {
   QGIS_PROTECT_QOBJECT_THREAD_ACCESS
 
+
+  if ( const QgsProviderMetadata *metadata = QgsProviderRegistry::instance()->providerMetadata( providerType() ) )
+  {
+    if ( metadata->providerCapabilities() & QgsProviderMetadata::ProviderCapability::LoadLayerMetadata )
+    {
+      try
+      {
+        const QgsLayerMetadata metadata { QgsProviderRegistry::instance()->loadLayerMetadata( providerType(), mDataSource, resultFlag ) };
+        if ( resultFlag )
+        {
+          mMetadata = metadata;
+          return tr( "Successfully loaded default layer metadata" );
+        }
+      }
+      catch ( QgsNotSupportedException &e )
+      {
+        resultFlag = false;
+        return e.what();
```

Comment:
shouldn't this cause a fallback to  loadNamedMetadata( metadataUri(), resultFlag ); ?

Reply:
I don't think so: if the provider does not support loading metadata we take the other branch which calls loadNamedMetadata on line 1326, this branch is taken by the multi-format providers (OGR) which in general advertises support for loading metadata but it may raise in case it doesn't for a particular format.

Reply:
@nyalldawson I am having second thoughts: you are right and it need to fallback. I'll change it.